### ISSUE TITLE
Merging datasets

### ISSUE THREAD

**Issue by evanfeinberg** [deepchem/deepchem#90]
After conducting hyperparameter search with K-fold CV, one common practice is to merge the different K acts (or, in the case of K=2 i.e. validation set, merge train and validation datasets) once hyperparameters are selected and then fitting on that combined dataset. It should be straightforward to add a method to the ```Dataset``` class to handle merging of datasets.

**Comment by rbharath** [comment]
This was added in #196.
